### ISSUE TITLE
Bluetooth Devices not showing - ScanningActivity

### ISSUE THREAD

**Issue by JalalJanKhan3336** [mazenrashed/Printooth#79]
**Greetings,** I'm using **Printooth** library to connect with a thermal printer namely [https://www.mobileworxs.com/products/mobile-printers/sewoo-lk-p41/](Sewoo IK P41). Sometimes it perfectly scans for available bluetooth devices but when the app is Offline it takes more time to scan or find available bluetooth devices. And at the end it doesn't show any available bluetooth device on ScanningActivity. Doesn't even show any error like "Scanning Timeout", "No device found".

In short, ScanningActivity sometimes doesn't work properly offline.

Please help me to resolve it.

Regards,

**Comment by ralphgabrielle** [comment]
Did you fix this? Im also facing the issue problem, any ideas would be a big help. Thanks

**Comment by adarshdhakad** [comment]
I am also facing the same issue 
then I turn on location now its woking fine

**Comment by adarsh-dhakad** [comment]
I am also facing the same issue
then I turn on location now its woking fine

**Comment by adarsh-dhakad** [comment]
I am also facing the same issue
then I turn on the location now its working fine

**Comment by salihinalfarizi** [comment]
already found solution for this issue...

you need request location permission, and set location device to active before run scanning bluetooth.
